microprofile/tests/tck/tck-opentracing/src/test/java/io/helidon/microprofile/opentracing/tck/UrlResourceProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.opentracing.tck;
+
+import java.lang.annotation.Annotation;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.test.spi.enricher.resource.ResourceProvider;
+
+/**
+ * TCKs use addition when creating URL for a client. The default Arquillian implementation returns url without the trailing
+ * /.
+ */
+public class UrlResourceProvider implements ResourceProvider {
+    @Override
+    public Object lookup(ArquillianResource arquillianResource, Annotation... annotations) {
+        try {
+            return URI.create("http://localhost:8080/").toURL();
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public boolean canProvide(Class<?> type) {
+        return type.isAssignableFrom(URL.class);
+    }
```

Comment:
This assignment is probably backwards. It states that _type_ is a supertype of _URL_. Also _URL_ is a final class.

Reply:
yep, will fix